woocommerce/woocommerce
# Install plugin-check and run it as a step in CI

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This installs the plugin-check plugin and executes the checks with that plugin.

Of the 5 categories of tests, only 1 passes (Accessibility) -- the other categories all have errors.  So I'm not sure what we want to do in terms of assertions and in terms of output for this.

I tried running the tests from the CLI and I tried running them in the browser (as they are here).  I like this implementation better because we can create better assertions using this -- for example, we could assert that no failures of a particular type occur.  We can also do simple assertions like "all tests passed"

Closes https://github.com/woocommerce/woocommerce/issues/47472



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check that CI passes
2. `pnpm env:restart && pnpm test:e2e-pw activate-and-setup/plugin-check`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
